src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,120 @@
+import classNames from 'classnames';
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  todo: Todo,
+  onDeleteTodo: (id: number) => void,
+  onUpdateTodo: (id: number, data: Partial<Todo>) => void,
+  isLoading: boolean,
+}
+
+export const TodoItem: React.FC<Props> = (
+  {
+    todo,
+    onDeleteTodo,
+    onUpdateTodo,
+    isLoading,
+  },
+) => {
+  const { title, completed, id } = todo;
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [todoTitle, setTodoTitle] = useState(title);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const cancelEditing = () => {
+    setIsEditing(false);
+    setTodoTitle(title);
+  };
+
+  const handleEsc = (event: KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      cancelEditing();
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!todoTitle) {
+      onDeleteTodo(id);
+    }
+
+    if (todoTitle === title) {
+      cancelEditing();
+    }
```

Comment:
you can write it in a shorter way 
(!todoTitle && onDeleteTodo(id))
(todoTitle === title && cancelEditing())

This is not a requirement to change, decided to let you know that you can do this as well